sql/core/src/test/scala/org/apache/spark/sql/execution/command/TestsV1AndV2Commands.scala

```diff
@@ -28,14 +28,21 @@ import org.apache.spark.sql.internal.SQLConf
 trait TestsV1AndV2Commands extends DDLCommandTestUtils {
   private var _version: String = ""
   override def commandVersion: String = _version
+  def runningV1Command: Boolean = commandVersion == "V1"
 
   // Tests using V1 catalogs will run with `spark.sql.legacy.useV1Command` on and off
   // to test both V1 and V2 commands.
   override def test(testName: String, testTags: Tag*)(testFun: => Any)
     (implicit pos: Position): Unit = {
     Seq(true, false).foreach { useV1Command =>
-      _version = if (useV1Command) "V1" else "V2"
+      def setCommandVersion(): Unit = {
+        _version = if (useV1Command) "V1" else "V2"
+      }
+      setCommandVersion()
       super.test(testName, testTags: _*) {
+        // Need to set command version inside this test function so that
+        // the correct command version is available in each test.
+        setCommandVersion()
```

Comment:
sorry I'm confused. We set the version right before `super.test(....)`, do you mean it's being set again during `super.test(....)`?

Reply:
This `def test` doesn't run `testFun` when it's invoked; it only registers the test). So by the time, `testFunc` is actually run, `_version` will always be set to "V2" (`useV1Command == false`). So we need to capture this inside the lambda that's passed into `super.test`.

Also, note that we need to call `setCommandVersion()` before calling `super.test` because `super.test` being called is utilizing the `commandVersion` to set the right test name:
https://github.com/apache/spark/blob/527e842ee6ed3bb1aabba61d02337ab81a56f87b/sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLCommandTestUtils.scala#L56-L57